autoPyTorch/evaluation/tae.py

```diff
@@ -62,34 +91,150 @@ def fit_predict_try_except_decorator(
     except (TypeError, OverflowError):
         return str(exit_status)
 
 
-class ExecuteTaFuncWithQueue(AbstractTAFunc):
+def _get_logger(logger_port: Optional[int], logger_name: str) -> Union[logging.Logger, PicklableClientLogger]:
+    if logger_port is None:
+        logger: Union[logging.Logger, PicklableClientLogger] = logging.getLogger(logger_name)
+    else:
+        logger = get_named_client_logger(name=logger_name, port=logger_port)
+
+    return logger
+
+
+def _get_origin(config: Union[int, str, Configuration]) -> str:
+    if isinstance(config, int):
+        origin = 'DUMMY'
+    elif isinstance(config, str):
+        origin = 'traditional'
+    else:
+        origin = getattr(config, 'origin', 'UNKNOWN')
+
+    return origin
+
+
+def _exception_handling(
+    obj: PynisherFunctionWrapperType,
+    queue: Queue,
+    info_msg: str,
+    info_for_empty: Dict[str, Any],
+    status: StatusType,
+    is_anything_exception: bool,
+    worst_possible_result: float
+) -> ProcessedResultsType:
+    """
+    Args:
+        obj (PynisherFuncWrapperType):
+        queue (multiprocessing.Queue): The run histories
+        info_msg (str):
+            a message for the `info` key in additional_run_info
+        info_for_empty (AdditionalRunInfo):
+            the additional_run_info in the case of empty queue
+        status (StatusType): status type of the running
+        is_anything_exception (bool):
+            Exception other than TimeoutException or MemorylimitException
+
+    Returns:
+        result (ProcessedResultsType):
+            cost, status, info, additional_run_info.
+    """
+    cost, info = worst_possible_result, None
+    additional_run_info: Dict[str, Any] = {}
+
+    try:
+        info = read_queue(queue)
+    except Empty:  # alternative of queue.empty(), which is not reliable
+        return cost, status, info, info_for_empty
+
+    result, status = info[-1]['loss'], info[-1]['status']
+    additional_run_info = info[-1]['additional_run_info']
+
+    _success_in_anything_exc = (is_anything_exception and obj.exit_status == 0)
+    _success_in_to_or_mle = (status in [StatusType.SUCCESS, StatusType.DONOTADVANCE]
+                             and not is_anything_exception)
+
+    if _success_in_anything_exc or _success_in_to_or_mle:
+        cost = result
+    if not is_anything_exception or not _success_in_anything_exc:
+        additional_run_info.update(
+            subprocess_stdout=obj.stdout,
+            subprocess_stderr=obj.stderr,
+            info=info_msg)
+    if is_anything_exception and not _success_in_anything_exc:
+        status = StatusType.CRASHED
+        additional_run_info.update(exit_status=_encode_exit_status(obj.exit_status))
+
+    return cost, status, info, additional_run_info
+
+
+def _process_exceptions(
+    obj: PynisherFunctionWrapperType,
+    queue: Queue,
+    budget: float,
+    worst_possible_result: float
+) -> ProcessedResultsType:
+    if obj.exit_status is TAEAbortException:
```

Comment:
I think this way of implementing the logic is a bit confusing. Could you rather maybe handle the exception first and then handle creating the `additional_run_info` based on that? Mainly I am struggling to find the use of `is_anything_exception`. Moreover, I also think that you are missing some info for example, `info_for_empty` in case of a `MEMOUT` previously also contained the memory limit as well. I'd suggest you to add both the `memory_limit` and `func_eval_time` which is available in `run_info.cutoff`.